webknossos-datastore/app/com/scalableminds/webknossos/datastore/models/requests/DataServiceRequests.scala

```diff
@@ -15,32 +16,41 @@ object DataServiceRequestSettings {
 }
 
 case class DataServiceDataRequest(
-    dataSource: DataSource, // null in VolumeTracings
+    dataSourceId: Option[DataSourceId], // None for volume tracings
     dataLayer: DataLayer,
     cuboid: Cuboid,
     settings: DataServiceRequestSettings
 ) {
   def isSingleBucket: Boolean = cuboid.isSingleBucket(DataLayer.bucketLength)
+  def mag: Vec3Int = cuboid.mag
+
+  // dataSource is None and unused for volume tracings. Insert dummy DataSourceId
+  // (also unused in that case, except for logging and bucket provider cache key)
+  def dataSourceIdOrVolumeDummy: DataSourceId = dataSourceId.getOrElse(DataSourceId("VolumeTracing", dataLayer.name))
 }
 
 case class DataReadInstruction(
     baseDir: Path,
-    dataSource: DataSource,
+    dataSourceId: DataSourceId, // Dummy value for volume tracings
```

Comment:
I initially got a little confused why the comment as it first read like "this field is a dummy value needed for volume tracing cases". To avoid this maybe something like

```suggestion
    dataSourceId: DataSourceId, // Dummy value in case of volume tracings
```